src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -15,7 +15,7 @@ namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
     /// <summary>
     /// Factory for <see cref="TagHelperDescriptor"/>s from <see cref="ITypeInfo"/>s.
     /// </summary>
-    public static class TagHelperDescriptorFactory
+    public class TagHelperDescriptorFactory
```

Comment:
Don't want to cache any higher than the `TagHelperDesignTimeDescriptorFactory` because higher level caching prevents change in `TagHelperDescriptor`s from being picked up. The documentation of design time descriptors on the other hand is less important and has a significant perf hit.